app.rb

```diff
@@ -0,0 +1,152 @@
+require_relative './person'
+require_relative './book'
+require_relative './student'
+require_relative './teacher'
+require_relative './rental'
+
+class App
+  attr_reader :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def menu
+    puts 'Welcome to School library App!\n\n'
+    puts 'Please choose an option by enter in a number: '
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit\n\n'
+    gets.chomp
+  end
+
+  def run
+    options = menu
+    case options
+    when '1'
+      list_all_books
+    when '2'
+      list_all_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_all_rentals
+    else
+      puts 'Thank you for using this app!'
+      exit
+    end
+  end
+
+  def list_all_books
+    if @books.size.positive?
+      puts 'Here are the books registered at the moment: '
+      @books.each_with_index { |book, index| puts "#{index}) Title: \"#{book.title}\" Author: #{book.author}" }
+    else
+      puts 'There are no books registered at the moment'
+    end
+    gets.chomp
+    run
+  end
+
+  def list_all_people
+    if @people.size.positive?
+      puts 'Here are the people registered at the moment: '
+      @people.each_with_index { |p, index| puts "#{index}) [#{p.class}] Name: #{p.name}, ID: #{p.id}, Age: #{p.age}" }
+    else
+      puts 'There are no people registered at the moment.'
+    end
+    gets.comp
+    run
+  end
+
+  def permision_options(my_char)
+    case my_char
+    when 'N'
+      false
+    when 'Y'
+      true
+    end
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp.capitalize
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(age, name, nil, specialization)
+    @people.push(teacher)
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp.capitalize
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp
+    student = Student.new(age, name, permission)
+    @people.push(student)
+  end
```

Comment:
- [ ]  I get an error while trying to  create a student, please checkout the screenshot below and kindly fix the issue.

![errorCreatingStudent](https://user-images.githubusercontent.com/65852621/186526127-52641076-2b3d-4601-97e5-dfce0a7de8fd.png)